packages/core/src/impl-store.ts

```diff
@@ -50,13 +50,16 @@ async function fetchOrDeployGeneric<T extends GenericDeployment>(
         );
       }
 
-      const stored = await getAndValidate(deployment, lens, provider);
-      const updated = merge ? await deploy() : await resumeOrDeploy(provider, stored, deploy);
+      const stored = deployment.get();
+      const updated = await resumeOrDeploy(provider, stored, deploy, lens.type, opts, deployment, merge);
```

Comment:
`stored` as an argument seems redundant given `deployment` is there.

Reply:
agreed, but `resumeOrDeploy` is exported and if we want to maintain backward compatibility in this version of core, we need to keep the `stored` parameter.